builtin/providers/aws/resource_aws_alb.go

```diff
@@ -0,0 +1,348 @@
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				ForceNew: true,
+				Optional: true,
+				Set:      schema.HashString,
+			},
+
+			"subnets": {
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				ForceNew: true,
+				Required: true,
+				Set:      schema.HashString,
+			},
+
+			"access_logs": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"bucket": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"prefix": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"enable_deletion_protection": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"idle_timeout": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  60,
+			},
+
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"zone_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"dns_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsAlbCreate(d *schema.ResourceData, meta interface{}) error {
+	elbconn := meta.(*AWSClient).elbv2conn
+
+	elbOpts := &elbv2.CreateLoadBalancerInput{
+		Name: aws.String(d.Get("name").(string)),
+		Tags: tagsFromMapELBv2(d.Get("tags").(map[string]interface{})),
+	}
+
+	if scheme, ok := d.GetOk("internal"); ok && scheme.(bool) {
+		elbOpts.Scheme = aws.String("internal")
+	}
+
+	if v, ok := d.GetOk("security_groups"); ok {
+		elbOpts.SecurityGroups = expandStringList(v.(*schema.Set).List())
+	}
+
+	if v, ok := d.GetOk("subnets"); ok {
+		elbOpts.Subnets = expandStringList(v.(*schema.Set).List())
+	}
+
+	log.Printf("[DEBUG] ALB create configuration: %#v", elbOpts)
+	var albArn string
+	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+		resp, err := elbconn.CreateLoadBalancer(elbOpts)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		if len(resp.LoadBalancers) != 1 {
+			return resource.NonRetryableError(fmt.Errorf(
+				"No loadbalancers returned following creation of %s", d.Get("name").(string)))
+		}
+
+		albArn = *resp.LoadBalancers[0].LoadBalancerArn
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(albArn)
+	log.Printf("[INFO] ALB ID: %s", d.Id())
+
+	return resourceAwsAlbUpdate(d, meta)
+}
+
+func resourceAwsAlbRead(d *schema.ResourceData, meta interface{}) error {
+	elbconn := meta.(*AWSClient).elbv2conn
+	albArn := d.Id()
+
+	describeAlbOpts := &elbv2.DescribeLoadBalancersInput{
+		LoadBalancerArns: []*string{aws.String(albArn)},
+	}
+
+	describeResp, err := elbconn.DescribeLoadBalancers(describeAlbOpts)
+	if err != nil {
+		if isLoadBalancerNotFound(err) {
+			// The ALB is gone now, so just remove it from the state
+			d.SetId("")
```

Comment:
Probably nitpick, but usually (in many other resources) we also send `[WARN]` log message to the log in such cases.